doc/docs/master/deploy-manage/manage/use-transactions-to-run-multiple-commands.md

```diff
@@ -0,0 +1,216 @@
+| `pachctl stop transaction` | Remove the currently active transaction from the local Pachyderm config file. The transaction remains in the Pachyderm cluster and can be resumed later. |
+| `pachctl resume transaction` | Set an already-existing transaction as the active transaction in the local Pachyderm config file. |
+| `pachctl delete transaction` | Deletes a transaction from the Pachyderm cluster. |
+| `pachctl inspect transaction` | Provides detailed information about an existing transaction, including which operations it will perform. By default, displays information about the current transaction. If you specify a transaction ID, displays information about the corresponding transaction. |
+
+## Supported Operations
+
+While there is a transaction object in the Pachyderm configuration
+file, all supported API requests append the request to the
+transaction instead of running directly. These supported commands include:
+
+```bash
+create repo
+delete repo
+start commit
+finish commit
+delete commit
+create branch
+delete branch
+```
+
+Each time you add a command to a transaction, Pachyderm validates the
+transaction against the current state of the cluster metadata and obtains
+any return values, which is important for such commands as
+`start commit`. If validation fails for any reason, Pachyderm does
+not add the operation to the transaction. If the transaction has been
+invalidated by changing the cluster state, you must delete the transaction
+and start over, taking into account the new state of the cluster.
+From a command-line perspective, these commands work identically within
+a transaction as without. The only difference is that you do not apply
+your changes until you run `finish transaction`, and a message that
+Pachyderm logs to `stderr` to indicate that the command was placed
+in a transaction rather than run directly.
+
+## Multiple Opened Transactions
+
+Some systems have a notion of *nested* transactions. That is when you
+open transactions within an already opened transaction. In such systems, the
+operations added to the subsequent transactions are not executed
+until all the nested transactions and the main transaction are closed.
+
+Pachyderm does not support such behavior. Instead, when you open a
+transaction, the transaction ID is written to the Pachyderm configuration
+file. If you begin another transaction while the first one is open, Pachyderm
+suspends the first transaction and overwrites the transaction ID in the
+configuration file. All operations that you add to the ensuing
+transactions will be executed as soon as you close those
+transactions. To resume the initial transaction, you need to run
+`pachctl resume transaction`.
+
+Because nested transactions are not supported, transactions cannot
+conflict with each other. Every time you add a command to a transaction,
+Pachyderm creates a blueprint of the commit and verifies that the
+command is valid.
```

Comment:
I know this is merged already, but this isn't clear to me for a couple of reasons:

1) AFAICS, whether transactions can nest is inconsequential to whether transactions can conflict.
2) What happens if two different users are running operations in their respective transactions at the same time? How are conflicting changes resolved? (e.g. if both transactions are creating a repo with the same name?)

Reply:
@Tryneus can you comment on the above?

Reply:
@ysimonson 

>  whether transactions can nest is inconsequential to whether transactions can conflict

Agreed - and while transactions can't necessarily conflict (for some definition of "conflict"), one transaction can invalidate another (e.g. by deleting a branch that the other transaction references).

> What happens if two different users are running operations in their respective transactions at the same time? How are conflicting changes resolved?

Transactions are at their core a batch of operations that are applied to the cluster state in one atomic operation.  Building a transaction over multiple commands has no effect on cluster state until the transaction is committed, although each command verifies that the entire batch is still valid by doing a dry run.

If two transactions create the same repo, they will both be valid until one transaction is committed, then the other transaction will get an error when attempting to commit it or add more commands into the transaction.